enterprise/management/commands/update_config_last_errored_at.py

```diff
@@ -0,0 +1,117 @@
+    'CANVAS': [CanvasEnterpriseCustomerConfiguration, CanvasLearnerDataTransmissionAudit],
+    'CANVAS_ASMT': [CanvasEnterpriseCustomerConfiguration, CanvasLearnerAssessmentDataTransmissionAudit],
+    'DEGREED': [DegreedEnterpriseCustomerConfiguration, DegreedLearnerDataTransmissionAudit],
+    'DEGREED2': [Degreed2EnterpriseCustomerConfiguration, Degreed2LearnerDataTransmissionAudit],
+    'GENERIC': [GenericEnterpriseCustomerPluginConfiguration, GenericLearnerDataTransmissionAudit],
+    'SAP': [SAPSuccessFactorsEnterpriseCustomerConfiguration, SapSuccessFactorsLearnerDataTransmissionAudit],
+}
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Command(IntegratedChannelCommandMixin, BaseCommand):
+    """
+    Management command which backfills missing audit record foreign keys.
+    """
+    help = _('''
+    Set error state for configurations.
+    ''')
+
+    def update_config_last_errored_at(self):
+        """
+        For each audit record kind (learner and content), find all the records in batch, and lookup
+        if they've had recent sync errors in the last day. If not, clear out the last_content_sync_errored_at
+        value associated with the configuration.
+        """
+        try:
+            has_learner_errors, has_content_errors = True, True
+            yesterday = datetime.now() - timedelta(days=1)
+            for channel_code, (ConfigModel, LearnerAuditModel) in MODELS.items():
+                configs = ConfigModel.objects.all().filter()
+                for config in configs:
+                    if config.last_sync_errored_at is None:
+                        continue
+                    customer_uuid = config.enterprise_customer.uuid
+                    plugin_id = config.id
+
+                    errored_learner_audits = LearnerAuditModel.objects.filter(
+                        created__date__gt=yesterday,
+                        status__gt=299,
+                        enterprise_customer_uuid=customer_uuid,
+                        plugin_configuration_id=plugin_id,
+                    )
+                    if not errored_learner_audits:
+                        config.last_learner_sync_errored_at = None
+                        has_learner_errors = False
+
+                    errored_content_audits = ContentMetadataItemTransmission.objects.filter(
+                        remote_created_at__gt=yesterday,
+                        enterprise_customer=config.enterprise_customer,
+                        integrated_channel_code=channel_code,
+                        plugin_configuration_id=plugin_id,
+                        api_response_status_code__gt=299
+                    )
+                    if not errored_content_audits:
+                        config.last_content_sync_errored_at = None
+                        has_content_errors = False
+
+                    if not has_learner_errors and not has_content_errors:
```

Comment:
why is `last_sync_errored_at` dependent on both no learner and content errors whereas the `last_learner_sync_errored_at` is only dependent on learner errors?

Reply:
So there's 3: last, last_learner, and last_content. We use them all in different contexts, but as you said last sync is dependent on both content and learner to show the user whats the most recent error regardless of type.